timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/ArrayLockEventSlidingWindow.java

```diff
@@ -18,22 +18,16 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Optional;
-import java.util.function.Supplier;
-
-import javax.annotation.concurrent.ThreadSafe;
 
 import com.google.common.math.LongMath;
 import com.google.common.primitives.Ints;
-import com.palantir.atlasdb.timelock.lock.watch.LockEventLogImpl.LockWatchCreatedEventReplayer;
-import com.palantir.lock.watch.LockWatchCreatedEvent;
 import com.palantir.lock.watch.LockWatchEvent;
+import com.palantir.logsafe.Preconditions;
 
-@ThreadSafe
 public class ArrayLockEventSlidingWindow {
```

Comment:
nit: Might be worth calling out `@NotThreadSafe` (it's really obvious, but worth noting)

Reply:
Done